src/include/api.h

```diff
@@ -156,6 +156,20 @@
 #define API_END_RET(s, ret) \
     API_END(s, ret);        \
     return (ret)
+
+#define API_END_STAT(s, ret, api)               \
+    if ((ret) != 0 && ((ret) != WT_NOTFOUND)) { \
+        WT_STAT_CONN_DATA_INCR(s, api##_error); \
+    }
+
```

Comment:
The uses of `API_END_STAT` without a trailing semicolon look awkward.  That could be fixed here by using the trick:
```
#define API_END_STAT(s, ret, api) do { \
  ....
  } while(0)
```
I think it's worth it to make the usage natural.  Otherwise, always doing a double take when reading the code.

Reply:
I think wrapping most or all of these macros in the `do { ... } while (0)` is important.

Reply:
I've added do whiles to a number of the macros, but left the cursor end ones for consistency, does that work @sueloverso?

Reply:
Yes, making it consistent with the others is fine. The reason, not just for looks and ease of semi-colons is to avoid any possibility of having a "trailing" if statement like this macro has. It is super important in this type of macro.